pkg/apis/autoscaling/v1alpha1/pa_types.go

```diff
@@ -155,15 +164,56 @@ func (pa *PodAutoscaler) ScaleBounds() (min, max int32) {
 	return
 }
 
-func (pa *PodAutoscaler) MetricTarget() (target int32, ok bool) {
+// Target returns the target annotation value or false if not present.
+func (pa *PodAutoscaler) Target() (target int32, ok bool) {
 	if s, ok := pa.Annotations[autoscaling.TargetAnnotationKey]; ok {
 		if i, err := strconv.Atoi(s); err == nil {
+			if i < 1 {
+				return 0, false
+			}
 			return int32(i), true
 		}
 	}
 	return 0, false
 }
 
+// Window returns the window annotation value or false if not present.
+func (pa *PodAutoscaler) Window() (window time.Duration, ok bool) {
+	if s, ok := pa.Annotations[autoscaling.WindowAnnotationKey]; ok {
+		d, err := time.ParseDuration(s)
+		if err != nil {
+			return 0, false
+		}
+		if d < time.Second*6 {
+			return 0, false
```

Comment:
Because `0` is the golang default value.  Returning `false` to indicate no valid value was provided.

Reply:
I think @markusthoemmes is asking about why it is 6 seconds.

Reply:
Ah, got it.  Moved the constant 6 to `register.go` and added comments to explain.  Essentially, the stable window should not be less than 6 seconds because as we approach 2 seconds, we are more likely to miss data points since we calculate scale every 2 seconds.  The panic window will always be smaller than the stable window, so 6 seems like a reasonable smallest value to me.